mep3_navigation/src/distance_angle/distance_angle_regulator.cpp

```diff
@@ -119,18 +143,22 @@ void DistanceAngleRegulator::odometry_callback(const nav_msgs::msg::Odometry::Sh
   const double distance_increment_angle = std::atan2(delta_y, delta_x);
 
   int sign = 1;
-  if (std::abs(robot_angle_ - distance_increment_angle) > 0.1) sign = -1;
+  if (std::abs(angle_normalize(robot_angle_ - distance_increment_angle)) > 0.1) sign = -1;
 
   robot_distance_ += sign * distance_increment;
 
   regulator_distance_.feedback = robot_distance_;
   regulator_angle_.feedback = robot_angle_;
 
-  regulator_distance_.reference = distance_profile_.update(time);
-  // We need to wrap the angle from -PI to PI
-  const double angle_ref = angle_normalize(angle_profile_.update(time));
-  regulator_angle_.reference = angle_ref;
+  /* RUCKIG */
+  motion_profile_result_ = motion_profile_->update(motion_profile_input_, motion_profile_output_);
 
+  if (motion_profile_result_ == ruckig::Result::Working) {
+    regulator_distance_.reference = motion_profile_output_.new_position[0];
+    regulator_angle_.reference = angle_normalize(motion_profile_output_.new_position[1]);
+    motion_profile_output_.pass_to_input(motion_profile_input_);
```

Comment:
```suggestion
    // TODO: Change to following once ruckig is updated.
    // motion_profile_output_.pass_to_input(motion_profile_input_);
    motion_profile_input_.current_position = motion_profile_output_.new_position;
    motion_profile_input_.current_velocity = motion_profile_output_.new_velocity;
    motion_profile_input_.current_acceleration = motion_profile_output_.new_acceleration;
```
Ja bih koristio verziju dostupnu u ROS-u, pa cim je azuriraju izmijenio bih i kod. Ovako ce malo biti mucno za studente da instaliraju

Reply:
Greska sa motion_command akcijom nastaje zbog internog bug-a unutar ruckiga (verzija is ros paketa). Treba najnoviji ruckig sa git-a.

Reply:
Moramo onda ostati na novoj verziji